google/cloud/internal/curl_wrappers.cc

```diff
@@ -243,26 +255,59 @@ std::size_t CurlAppendHeaderData(CurlReceivedHeaders& received_headers,
   return size;
 }
 
+std::string DebugInfo(char const* data, std::size_t size) {
+  return absl::StrCat("== curl(Info): ", absl::string_view{data, size});
+}
+
+std::string DebugRecvHeader(char const* data, std::size_t size) {
+  return absl::StrCat("<< curl(Recv Header): ", absl::string_view{data, size});
+}
+
+std::string DebugSendHeader(char const* data, std::size_t size) {
+  auto header = absl::string_view{data, size};
+  auto const match = absl::string_view{": Bearer "};
+  auto const limit = match.size() + 32;
+  auto const bearer_pos = header.find(match);
+  if (bearer_pos != std::string::npos) {
+    auto const nl_pos = header.find('\n', bearer_pos);
+    auto const n = (std::min)(limit, nl_pos - bearer_pos);
+    return absl::StrCat(">> curl(Send Header): ", header.substr(0, bearer_pos),
+                        header.substr(bearer_pos, n), "...<truncated>...",
+                        header.substr(nl_pos), "\n");
```

Comment:
This seems to append a trailing `"\n"` that we wouldn't get in the `bearer_pos == std::string::npos` case.

Reply:
I think you meant `nl_pos == std::string::npos`, I fixed that.  Otherwise, this is in the body of a `if (bearer_pos != std::string::npos)` ?